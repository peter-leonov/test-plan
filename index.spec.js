// abusing Dependency Injection to make side effects explicit
function incrementComments(fetch, log, save, postId) {
    const post = fetch(`/posts/${postId}`);
    if (!post) return;
    log('comment.inc', postId);
    post.comments++;
    save(post);
}

const { plan } = require('.');
describe('incrementComments', () => {
    it('increments existing posts', () => {
        const p = plan();
        incrementComments(
            p.fetch(() => ({
                comments: 1,
            })),
            p.log(),
            p.save(),
            123,
        );
        expect(p()).toMatchSnapshot();
    });

    it('ignores non existing posts', () => {
        const p = plan();
        incrementComments(p.fetch(() => null), p.log(), p.save(), 123);
        expect(p()).toMatchSnapshot();
    });
});
