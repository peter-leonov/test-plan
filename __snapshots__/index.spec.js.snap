// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`incrementComments ignores non existing posts 1`] = `
Array [
  Array [
    "call",
    "fetch",
    Array [
      "/posts/123",
    ],
  ],
]
`;

exports[`incrementComments increments existing posts 1`] = `
Array [
  Array [
    "call",
    "fetch",
    Array [
      "/posts/123",
    ],
  ],
  Array [
    "call",
    "log",
    Array [
      "comment.inc",
      123,
    ],
  ],
  Array [
    "call",
    "save",
    Array [
      Object {
        "comments": 2,
      },
    ],
  ],
]
`;
